openwisp_controller/config/tests/test_template.py

```diff
@@ -466,6 +466,20 @@ def test_required_templates(self):
             self.assertTrue(config.templates.filter(pk=t1.pk).exists())
             self.assertTrue(config.templates.filter(pk=t2.pk).exists())
 
+        with self.subTest('Required templates honours backend of config'):
+            config.backend = 'netjsonconfig.OpenWisp'
+            config.save()
+            self.assertEqual(config.templates.count(), 3)
+
+            # This should not raise an exception since backends of
+            # t5 and config is not same. Also t5 should not be added back
+            config.templates.remove(t5)
+            self.assertEqual(config.templates.count(), 2)
```

Comment:
is this just to be extra sure (which is ok) or did you expect this to fail without the fix (which would not be ok and would have to be looked into)?

Reply:
Upon closer look, I realised `t5` was not a required template. Fixed this bogus test. Thank you very much Federico for catching this.